### DISCUSSION TITLE
Ts error on type annotation

### DISCUSSION THREAD

**Discussion by ImamJanjua** [pmndrs/zustand#2554] [Q&A] [answered]
https://github.com/pmndrs/zustand/assets/137432044/4c50964d-6290-4179-b169-0ce60bc91b14

**Comment by charkour** [comment]
Hi @ImamJanjua, thanks for the report. Could you provide a written account of what the current behavior is and what the expected behavior is? Thank you

**Comment by ImamJanjua** [comment]
@charkour the video shows that there are no errors, till i add the view state which should be either "day" or "week" and the action setView which accepts "day" or "week" and sets the view based on input.

**Comment by dbritto-dev** [comment]
@ImamJanjua your action types are wrong, there's no `setView`, and you should type your initial state too

**Reply by dbritto-dev** [reply, marked as answer]
<img width="933" alt="image" src="https://github.com/pmndrs/zustand/assets/1697714/40fd4dac-0e47-4d1b-b707-4435f19eb586">
